var/spack/repos/builtin/packages/hypar/package.py

```diff
@@ -18,8 +18,8 @@ class Hypar(AutotoolsPackage):
     """
 
     homepage = "http://hypar.github.io/"
-    url = "https://bitbucket.org/deboghosh/hypar/get/v4.1.tar.gz"
-    git = "https://bitbucket.org/deboghosh/hypar.git"
+    url = "https://github.com/debog/hypar/archive/refs/tags/v4.1.tar.gz"
```

Comment:
Are the checksums unchanged for listed versions?

Reply:
I am not sure. The contents should be exactly the same. How do I find out?

Reply:
Unfortunately github creates its own archives, and the checksums can be dependent on local settings (e.g. compression level). In this case it seems the checksum is different on github.
```
$ curl -L https://github.com/debog/hypar/archive/refs/tags/v4.1.tar.gz | sha256sum 
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  181M    0  181M    0     0  12.8M      0 --:--:--  0:00:14 --:--:-- 14.3M
b3bfc6da28d78e2cc89868a35990617e4f77521b68911772887c2f8d0b1fec21  -
```
Only when you explicitly upload a file as a release artifact can you be sure that it won't change. But, then you obscure what is in those files since they don't agree with what's visible in the repository (see xz attack for what that then allows).

Reply:
Okay, thanks for explaining. How do I update the checksum information? Is there another file I need to update?

Reply:
Updating the checksum in the package.py file is sufficient.

Reply:
Oh yes, it's there a couple of lines below! :-) Updated it.